datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaConnector.java

```diff
@@ -160,25 +158,20 @@ public Thread createTaskThread(AbstractKafkaBasedConnectorTask task) {
     return t;
   }
 
+  // The _runningTasks must be locked before calling this method
```

Comment:
done - I was debating about this too, and then decided I'd make this change as part of the next change to force cancel task in a separate thread during onAssignmentChange(). Easy enough to do it as part of this change itself though.

I decided to remove the need of _runningTasks in these functions rather than making them have synchronized blocks too.

Reply:
Awesome. Thank you.